ingest-server/src/main/java/cz/o2/proxima/server/IngestServer.java

```diff
@@ -158,13 +156,15 @@ public void onCompleted() {
             if (res > 0) {
               synchronized (inflightRequest) {
                 try {
-                  inflightRequest.wait();
+                  while (inflightRequest.get()>0) {
```

Comment:
Minor: coding style

Reply:
And regarding functionality - we should probably not wait until the `inflightRequests` are empty, but rather while we can enqueue the request, because then it can result in faster sending of the request.

Reply:
The previous comment can be ignored, it looks fine.